### PR TITLE
Pin numba until such time @overload can be implemented

### LINKED ISSUE #280: Pin release version of numba to < 0.59.0
Codex version: 0.3.4

Raised downstream by crystalball. I see generated_jit is being removed in the near future so I'm marking this and make a patch release of codex to ensure the downstream packages don't start failing -- I don't think there is time now to modify the codebase to use @overload. Per docs we should pin to to below that version.

### PR DESCRIPTION
**Pull request by bennahugo**
Closes #280 

- [ ] Tests added / passed

  ```bash
  $ py.test -v -s africanus
  ```

  If the pep8 tests fail, the quickest way to correct
  this is to run `autopep8` and then `flake8` and
  `pycodestyle` to fix the remaining issues.

  ```
  $ pip install -U autopep8 flake8 pycodestyle
  $ autopep8 -r -i africanus
  $ flake8 africanus
  $ pycodestyle africanus
  ```

- [ ] Fully documented, including `HISTORY.rst` for all changes
      and one of the `docs/*-api.rst` files for new API

  To build the docs locally:

  ```
  pip install -r requirements.readthedocs.txt
  cd docs
  READTHEDOCS=True make html
  ```

### CODE REVIEW

**Comment by sjperkins** [comment]
Thanks for looking at his @bennahugo. Pinning numba seems to be a sensible interim measure.

> Hmm tests are currently not working. Setting NUMBA_NRT_STATS appears not to enable NRT counters. Will need further digging

 One thing that immediately springs to mind is the following fixture https://github.com/ratt-ru/codex-africanus/blob/9f6836a5402673bedda2035956a3786dc1ddc61b/africanus/conftest.py#L9-L17 which was designed to check for NRT allocations in the Fused RIME. If the NRT interface changed in numba, this might be the culprit for the test failures.

I'd leave it for Monday though!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -79,7 +79,7 @@ jobs:
         run: pipenv graph
 
       - name: Run complete test suite
-        run: pipenv run py.test --flake8 -s -vvv africanus/
+        run: NUMBA_NRT_STATS=1 pipenv run py.test --flake8 -s -vvv africanus/
```
Comment (sjperkins) [inline .github/workflows/ci.yml:82]: I don't think `NUMBA_NRT_STATS=1` gets inherited by the  virtual environment created by pipenv


```suggestion
        run: pipenv run NUMBA_NRT_STATS=1 py.test --flake8 -s -vvv africanus/
```

In a fresh python 3.9 virtual environment the test suite is running:

```bash
$ NUMBA_NRT_STATS=1 py.test --flake8 -s -vvv africanus/
```

**Comment by sjperkins** [comment]
Related to:

- https://github.com/numba/numba/pull/8235

**Comment by bennahugo** [comment]
Superceeded by #284

**Pull request closed without merging**